synapse/lib/stormlib/model.py

```diff
@@ -259,35 +258,70 @@ def __init__(self, form, path=None):
 
     def getObjLocals(self):
         return {
-            'prop': self._getFormProp
+            'prop': self._getFormProp,
+            'name': self._propFormName,
         }
 
+    @property
+    def _propFormName(self):
+        '''
+        The name of the Form.
+        '''
+        return self.valu.name
+
     def _ctorFormType(self, path=None):
         return ModelType(self.valu.type, path=path)
 
     def _getFormProp(self, name):
+        '''
+        Get a Property on the Form.
+
+        Args:
+            name (str): The property to retrieve.
+
+        Returns:
+            ModelProp instance if hte property if present on the Form, or ``$lib.null``.
+        '''
         prop = self.valu.prop(name)
         if prop is not None:
             return ModelProp(prop)
 
 @s_stormtypes.registry.registerType
 class ModelProp(s_stormtypes.Prim):
-
+    '''
+    Implemen the Storm API for a Property.
+    '''
     def __init__(self, prop, path=None):
 
         s_stormtypes.Prim.__init__(self, prop, path=path)
 
-        # Todo: Plumb name and full access via a @property and implement getObjLocals
-        self.locls.update({
-            'name': prop.name,
-            'full': prop.full,
-        })
-
         self.ctors.update({
             'form': self._ctorPropForm,
             'type': self._ctorPropType,
         })
 
+        self.locls.update(self.getObjLocals())
+
+    def getObjLocals(self):
+        return {
+            'full': self._propFull,
+            'name': self._propName,
+        }
+
+    @property
+    def _propName(self):
```

Comment:
Not 100% sure i'm digging the fact that every deref property access ends up incurring the cost of an additional function call frame with this pattern.  Not against it, but lets discuss :)